[docs] Add a Review section with overview in Home (#26064)
expo/expo: docs/pages/review/overview.mdx

<Collapsible summary="Can I use Expo Go for reviewing releases?">

Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.

</Collapsible>

### Get started

<Video
  file="review/eas-update-01.mp4"
  spaceAfter={70}
  caption="When an update is published, you can use the Expo dashboard to generate and share links with your team."
/>

<BoxLink
  title="Get started with EAS Update"
  description={
    <>
      Learn how to get started using <CODE>expo-updates</CODE> library and use EAS Update in your
      project.
    </>
  }
  href="/eas-update/getting-started/"
  Icon={LayersTwo02Icon}
/>

### `expo-dev-client` library

<Video
  file="review/eas-update-02.mp4"
  spaceAfter={70}
  caption="Using the development build on a device allows you to fetch and view updates directly."
/>

<BoxLink
  title={
    <>
      Use <CODE>expo-dev-client</CODE> with EAS Update
    </>
  }
  description={
    <>

>       Learn how to use <CODE>expo-dev-client</CODE> in your project to launch different app versions
>       and preview a published update inside a development build. This library also supports loading
>       published EAS updates through channels.

Author: ```suggestion
      Learn how to use <CODE>expo-dev-client</CODE> in your project to launch different app versions
      and preview a published update inside a development build.
```

Internal distribution is a strategy for sharing an app with a limited audience. It is used for testing purposes within a team or organization or a selected group of testers. Android and iOS provide different ways to install a build of your application directly on a device.

Google Play beta provides both closed and open tracks. You can invite specific users to test your app in the closed beta. In the open beta, you can make your app available to anyone participating in the beta program.

Apple offers ad hoc provisioning profile to distribute your app to test on devices registered to your Apple Developer account. You can also use TestFlight to share your app with up to 10,000 external testers by [uploading your app](/submit/ios/).

However, both of these methods are time-consuming and limiting. EAS Build can help you overcome these limitations by providing a shareable URL for your team to get it up and running.

<BoxLink
  title="How to set up an internal distribution build"
  description="Learn how EAS Build provides shareable URLs for your builds with your team for internal distribution."
  href="/build/internal-distribution/"
/>

## Using EAS Update with development builds

> Development builds are highly recommended for a robust and flexible solution, especially when planning to deploy your app to production. They provide a comprehensive development environment that can be further configured to generate previews for your app during the review stage.

Reviewer: ```suggestion
You can use [development builds](/develop/development-builds/introduction/) to load previews of your app during the review stage by publishing the preview with EAS Update.
```

> Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.
> 
> You can use a development build with EAS Update and see the getting started guide for further instructions to set up the `expo-updates` library and learn how to create a link between a build and an update.
> 

Reviewer: ```suggestion
Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.
```

Reviewer: also, maybe move this Expo Go part inside of a collapsible? like "Can I use Expo Go for reviewing releases?"

<BoxLink
  title="Get started with EAS Update"
  description="Learn how to get started using expo-updates library and use EAS Update in your project."
  href="/eas-update/expo-dev-client/"
/>

EAS Update allows you to push updates to iterate faster with your team. You can publish a new update for your team to preview. This process allows any team member or reviewer with a registered device to review the latest version of your app by scanning a QR code.

<BoxLink
  title="Develop with EAS Update"
  description="Learn how to iterate faster locally or with your team."
  href="/eas-update/develop-faster/"
/>

If you have the `expo-dev-client` library installed in your project, your team can launch different app versions and preview a published update inside the development build. It supports loading published EAS updates through channels.

> 

Reviewer: maybe we should reword this slightly and move it to the "description" field for the box link below?

<BoxLink
  title="Use expo-dev-client with EAS Update"
  description="Learn how to use expo-dev-client library to preview a published EAS Update inside a development build."
  href="/eas-update/expo-dev-client/"
/>

You can also use GitHub Actions to automate the process of publishing updates. It makes deploying updates consistent and fast, leaving you more time to develop your app.

> 

Reviewer: same re: moving to box link description

  href="/build/internal-distribution/"
  Icon={BuildIcon}
/>

## Using EAS Update with development builds

You can use [development builds](/develop/development-builds/introduction/) to load previews of your app during the review stage by publishing the preview with EAS Update.

<Collapsible summary="Can I use Expo Go for reviewing releases?">

Even though Expo Go is an open-source sandbox that can be good for previewing isolated prototypes on Android and iOS, it is not intended for production apps. It should be avoided during the preview process of your app.

</Collapsible>

<BoxLink
  title="Get started with EAS Update"
  description={
    <>
      Learn how to get started using <CODE>expo-updates</CODE> library and use EAS Update in your
      project.
    </>
  }
  href="/eas-update/getting-started/"
  Icon={LayersTwo02Icon}
/>

<BoxLink
  title="Develop with EAS Update"
  description="Learn how to use EAS Update to push changes to iterate faster with your team. This process allows any team member or reviewer with a registered device to review the latest version of your app by scanning the QR code."
  href="/eas-update/develop-faster/"
  Icon={LayersTwo02Icon}

> />

Reviewer: this link may not be particularly useful anymore after https://github.com/expo/expo/pull/27461 - wdyt @wschurman / @amandeepmittal?

Author: I just saw Will's PR and got the context. Will remove this link.

Author: Done ✅
